overlord/devicestate/systems.go

```diff
@@ -426,3 +429,49 @@ func createSystemForModelFromValidatedSnaps(model *asserts.Model, label string,
 
 	return recoverySystemDir, nil
 }
+
+// ModelAndGadgetInfoFromSeed return the model assertion and gadget details
+// for the given system label.
+func (m *DeviceManager) ModelAndGadgetInfoFromSeed(wantedSystemLabel string) (*asserts.Model, *gadget.Info, error) {
```

Comment:
I partly forgot about DeviceManager.Systems, I wonder if this should be called SystemAndGadgetInfo and return a System instead of just a model ?